src/course-unit/data/slice.js

```diff
@@ -87,6 +89,15 @@ const slice = createSlice({
         fetchUnitLoadingStatus: payload.status,
       };
     },
+    fetchCourseVerticalChildrenDataSuccess: (state, { payload }) => {
+      state.courseVerticalChildren = payload;
+    },
+    updateLoadingCourseVerticalChildrenDataStatus: (state, { payload }) => {
+      state.loadingStatus = {
+        ...state.loadingStatus,
+        courseVerticalChildrenLoadingStatus: payload.status,
+      };
```

Comment:
```suggestion
      state.loadingStatus.courseVerticalChildrenLoadingStatus = payload.status;
```
I am not sure, but this can work according to `immer` library.

Reply:
According to the docs it should work

Reply:
If yes, we need to refactor all previously created methods.
I think, we should leave as is to stick to code consistency.

Reply:
Done